TACC/Core-Styles
# fix: figure image shrunk if figure was not aligned

## Overview

**Problem**: Shrink-wrapped figure shrinks image that has `max-width: 100%`.

**Solution**: Only apply `width: min-content` to figure if it is aligned.

**Origin**: #114 @ [`src/lib/_imports/elements/html-elements.css` line 59](https://github.com/TACC/Core-Styles/pull/114/files#diff-380b6e87b7b95fe11c34205c7c6906541a722cbe3c877498791d1268e9b5d4faR59)

</details>

## Related

- fixes bug from #114
- implements figure alignment how #115 should have

## Changes

- **fix**: an `<img>` in a `<figure>` was shrunk if `<figure>` was not aligned c30de99
- _**feat**: @figure partial can give wide and tall images 4ca0897_
- _**feat**: "HTML Elements" demo adds a figure with a variable image width a4c55a0 (and 2f721df)_

## Testing

1. Open http://localhost:3000/components/detail/html-elements.
2. Scroll to the two figures.
3. Verify neither figures caption wraps at image width.
4. Verify neither image is shrunk (less than what its markup and styles dictate).

## UI

| default figures<br><sup>(no image shrink)</sup> | aligned figures<br><sup>(unchanged)</sup> | aligned blockquotes<br><sup>(unchanged)</sup> |
| - | - | - |
| ![default figures (no wrap now)](https://user-images.githubusercontent.com/62723358/215917903-935f9f2c-fbff-46a5-b90f-6ccdd6ef0eec.png) | ![aligned figures (unchanged)](https://user-images.githubusercontent.com/62723358/215917895-5d7dd195-3c48-4f59-b6df-3483e35339c0.png) | ![aligned blockquotes (unchanged)](https://user-images.githubusercontent.com/62723358/215917899-a978185d-8c19-4796-bb56-0fd1cec47741.png) |
